src/backend/joanie/core/api/client.py

```diff
@@ -426,3 +427,97 @@ def download(self, request, pk=None):  # pylint: disable=no-self-use, invalid-na
         response["Content-Disposition"] = f"attachment; filename={pk}.pdf;"
 
         return response
+
+
+class OrganizationViewSet(
+    mixins.ListModelMixin,
+    mixins.RetrieveModelMixin,
+    viewsets.GenericViewSet,
+):
+    """
+    API ViewSet for all interactions with organizations.
+
+    GET /api/organizations/:organization_id
+        Return list of all organizations related to the logged-in user or one organization
+        if an id is provided.
+    """
+
+    lookup_field = "pk"
+    pagination_class = Pagination
+    permission_classes = [permissions.IsAuthenticated]
+    serializer_class = serializers.OrganizationSerializer
+
+    def get_queryset(self):
+        """
+        Custom queryset to get user organizations
+        """
+        user = models.User.update_or_create_from_request_user(
+            request_user=self.request.user
+        )
+        return models.Organization.objects.filter(accesses__user=user)
+
+
+class OrganizationAccessViewSet(
+    mixins.CreateModelMixin,
+    mixins.DestroyModelMixin,
+    mixins.ListModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.UpdateModelMixin,
+    viewsets.GenericViewSet,
+):
+    """
+    API ViewSet for all interactions with organization accesses.
+
+    GET /api/organization-accesses/:organization_access_id
+        Return list of all organization accesses related to the logged-in user or one
+        organization access if an id is provided.
+
+    POST /api/organization-accesses/ with expected data:
+        - user: str
+        - organization: str
+        - role: str [owner|admin|member]
+        Return newly created organization access
+
+    PUT /api/organization-accesses/<organization_access_id>/ with expected data:
+        - role: str [owner|admin|member]
+        Return updated organization access
+
+    DELETE /api/organization-accesses/<organization_access_id>/
+        Delete targeted organization access
+    """
+
+    lookup_field = "pk"
+    pagination_class = Pagination
```

Comment:
There are always one access for the couple organization/user. Maybe Pagination is useless here ?

Reply:
but if I'm an admin I cansee maybe accesses for the organization. Right?

Reply:
But currently, for the list view we are filtering access for the authenticated user.

Reply:
No it's tricky but it's a bit different:
```
        if self.action == "list":
            queryset = queryset.filter(
                organization__accesses__user__username=self.request.user.username,
            ).distinct()
```
I'm filtering accesses that are linked to an organization THAT has an access for the logged-in user!